components/raftstore/src/store/worker/compact.rs

```diff
@@ -169,6 +200,49 @@ where
                 }
                 Err(e) => warn!("check ranges need reclaim failed"; "err" => %e),
             },
+            Task::TTLCheckAndCompact { start_key, end_key } => {
+                let current_ts = UnixSecs::now().into_inner();
+                let range = Range::new(&start_key, &end_key);
+                let collection = match self
+                    .engine
+                    .get_properties_of_tables_in_range(CF_DEFAULT, &[range])
+                {
+                    Ok(v) => v,
+                    Err(e) => {
+                        error!(
+                            "execute ttl compact files failed";
+                            "range_start" => log_wrappers::Value::key(&start_key),
+                            "range_end" => log_wrappers::Value::key(&end_key),
+                            "err" => %e,
+                        );
+                        return;
+                    }
+                };
+
+                if collection.is_empty() {
+                    return;
+                }
+
+                let mut files = Vec::new();
+                for (file_name, v) in collection.iter() {
+                    let prop = match TTLProperties::decode(&v.user_collected_properties()) {
+                        Ok(v) => v,
+                        Err(_) => continue,
+                    };
+                    if prop.max_expire_ts <= current_ts {
```

Comment:
I think we do not need to compact every file which concludes an expired key, because if this region is updated frequently, the file may be compacted automatic. We only need to compact files which has not been compacted for a long time. Maybe we could add the time one or two hour?

Reply:
note that it's comparing to the `max_expire_ts`, so the file is compacted only when all keys are expired. Also, considering the interval of checking SSTs' TTL may be hours(24h by default), I don't think it won't be compacted so frequently.

Reply:
But if user write a log of keys today. The time in tomorrow will trigger a lot of manual compaction. However, we only need to wait a hour, maybe most of them will be compacted automatic. And then we only need to compact few regions.

Reply:
why most of them will be compacted automatically tomorrow?